libbpf-rs/src/perf_buffer.rs

```diff
@@ -282,6 +282,26 @@ impl<'b> PerfBuffer<'b> {
             unsafe { libbpf_sys::perf_buffer__buffer_fd(self.ptr, buf_idx as libbpf_sys::size_t) };
         util::parse_ret_i32(ret)
     }
+
+    pub fn buffer_buffer(&self, buf_idx: usize) -> Result<&mut [u8]> {
```

Comment:
I don't believe this API is sound. If you call it multiple times with the same index you get multiple mutable references to the same area of memory, right?

Also, documentation it missing.

Reply:
and the name should be just "buffer"?

Reply:
Renamed to buffer, as for would you get multiple mutable references to the same area is true. i'll add a comment about that.